### ISSUE TITLE
varargs + optional params doesn't work

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#14436]
varargs + optional params doesn't work

### Example
```nim
when true:
  template fun*(args: varargs[untyped], ok = 123) =
    echo args
    echo ok
  fun("ab", "cd") # ok
  fun("ab", "cd", ok = 421) # error: Error: type mismatch: got <string, string, ok: int literal(421)>

when false: # uncomment for 2nd error
  template fun*(ok = 123, args: varargs[untyped]) =
    echo args
    echo ok
  fun("ab", "cd") #  Error: type mismatch: got <string, string>
  fun(ok = 4, "ab", "cd") # ok
```

### Current Output
```
error: Error: type mismatch: got <string, string, ok: int literal(421)>
```

### Expected Output
```
works
```

### Possible Solution
* sigmatch should allow binding arguments by named parameters even in presence of a varargs param

### Additional Information
* https://github.com/nim-lang/Nim/issues/9996 but different
* c61f7466292cb19c910f4a2f2976f738796e6bdf 1.3.5

**Comment by alaviss** [comment]
>  * sigmatch should allow binding arguments by named parameters even in presence of a varargs param

I don't think this is a good idea. Instead we might want to make it so that `varargs[untyped]` and `varargs[typed]` must be the last parameter.

This should eliminate any surprises that might happen if some API for example, add a new optional parameter called `copy` at the end, and now an AST like `copy = false` is now treated as setting the optional parameter.
